test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Json.Tests.ps1

```diff
@@ -156,4 +156,48 @@ Describe 'ConvertTo-Json' -tags "CI" {
         $actual = ConvertTo-Json -Compress -InputObject $obj
         $actual | Should -Be '{"Positive":18446744073709551615,"Negative":-18446744073709551615}'
     }
+
+    It "Should convert objects to JSON as Stream" {
+        $objects = @(
+            [pscustomobject]@{ Name = "Object1"; Value = 10 },
+            [pscustomobject]@{ Name = "Object2"; Value = 20 },
+            [pscustomobject]@{ Name = "Object3"; Value = 30 }
+        )
+        $returnObject = $objects | ConvertTo-Json -Compress -AsStream
+        $returnObject -is [System.Array] | Should -BeTrue
+        $returnObject | Should -HaveCount 3
+        $returnObject[0] | Should -Be '{"Name":"Object1","Value":10}'
+        $returnObject[1] | Should -Be '{"Name":"Object2","Value":20}'
+        $returnObject[2] | Should -Be '{"Name":"Object3","Value":30}'
+    }
+
+    It "Should convert nested array to JSON as Stream" {
+        $objects = ((1,2),3,(4,5))
+        $returnObject = $objects | ConvertTo-Json -Compress -AsStream
+        $returnObject -is [System.Array] | Should -BeTrue
+        $returnObject | Should -HaveCount 3
+        $returnObject[0] | Should -Be '[1,2]'
+        $returnObject[1] | Should -Be '3'
+        $returnObject[2] | Should -Be '[4,5]'
+    }   
+
+    It "StopProcessing as stream should succeed" -Pending:$true {
+        $ps = [PowerShell]::Create()
+        $null = $ps.AddScript({
+            $obj = [PSCustomObject]@{P1 = ''; P2 = ''; P3 = ''; P4 = ''; P5 = ''; P6 = ''}
```

Comment:
Instead of making a complex object you could potentially look at creating a `ScriptProperty` with a sleep in it

```powershell
$obj = [PSCustomObject]@{} | Add-Member -MemberType ScriptProperty -Name Prop -Value {
    Start-Sleep -Seconds 5
    'value'
} -PassThru
$obj | ConvertTo-Json
```